sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,708 @@
+        labels = np.zeros(X.shape[0], dtype=np.intc)
+
+        labels_left = np.arange(X.shape[0])
+
+        centroids = []
+
+        init_org = init
+
+        for n_iter in range(self.n_clusters):
+
+            # If init array is provided -
+            # Take only part of init that is dedicated for that part of bisect
+            if hasattr(init, '__array__'):
+                init = init_org[n_iter: n_iter + 2].copy()
+
+            # Perform Bisection
+            centers, _labels = self._bisect(data_left, init, weights_left,
+                                            random_state)
+
+            # Check SSE (Sum of Squared Errors) of each of computed centroids.
+            # SSE is calculated with distances between data points
+            # and assigned centroids
+            errors = self._compute_bisect_errors(data_left, centers, _labels,
+                                                 weights_left)
+
+            lower_sse_index = 0 if errors[0] < errors[1] else 1
+
+            higher_sse_index = 1 if lower_sse_index == 0 else 0
+
+            # Centroid with lower SSE is added to result centroids.
+            centroids.append(centers[lower_sse_index])
+
+            # Label data with lower SSE
+            labels[labels_left[(_labels == lower_sse_index)]] = n_iter
+
+            # Data with label that has higher SSE will be further processed
+            indexes_left = (_labels == higher_sse_index)
+
+            if self.verbose:
+                print(f"Centroid Found: {centers[lower_sse_index]}")
+
+            if len(centroids) + 1 == self.n_clusters:
+                # Desired number of cluster centers is reached so centroid
+                # with higher SSE would not be split further
+                centroids.append(centers[higher_sse_index])
+                labels[labels_left[indexes_left]] = n_iter + 1
+
+                if self.verbose:
+                    print(f"Centroid Found: {centers[higher_sse_index]}")
+                break
+
+            # Pass for further split data, weights and labels left
+            data_left = data_left[indexes_left]
+            weights_left = weights_left[indexes_left]
+            labels_left = labels_left[indexes_left]
+
+        self.labels_ = labels
+        self.cluster_centers_ = np.asarray(centroids)
+
+    def _bisect_by_biggest_sse(self, X, init, random_state, sample_weight):
+        """ Performs Bisecting K-Means, which splits always cluster with
+         biggest SSE from all calculated
+
+            .. note:: All of passed parameters must be pre-calculated
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training instances to cluster.
+
+        init : {'k-means++', 'random'}, callable or ndarray of shape \
+                (n_clusters, n_features)
+                Method for initialization.
+
+        random_state : int, RandomState instance or None, default=None
```

Comment:
The docstring needs to be updated. There is no default, and it is always a `RandomState` instance here.

Reply:
Fixed with next commit